sdk/servicebus/azure-messaging-servicebus-track1-perf/src/main/java/com/azure/messaging/servicebus/perf/ReceiveAndDeleteMessageTest.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus.perf;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.perf.core.ServiceBusStressOptions;
+import com.azure.messaging.servicebus.perf.core.ServiceTest;
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.Message;
+import com.microsoft.azure.servicebus.ReceiveMode;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Performance test.
+ */
+public class ReceiveAndDeleteMessageTest extends ServiceTest<ServiceBusStressOptions> {
+    private final ClientLogger logger = new ClientLogger(ReceiveAndDeleteMessageTest.class);
+    private List<Message> messages = new ArrayList<>();
+
+    /**
+     * Creates test object
+     * @param options to set performance test options.
+     */
+    public ReceiveAndDeleteMessageTest(ServiceBusStressOptions options) {
+        super(options, ReceiveMode.RECEIVEANDDELETE);
+    }
+
+    @Override
+    public Mono<Void> cleanupAsync() {
+        try {
+            sender.close();
+        } catch (ServiceBusException e) {
+            throw logger.logExceptionAsWarning(new RuntimeException(e));
+        }
+        return Mono.empty();
+    }
+
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        try {
+            sender.close();
+        } catch (ServiceBusException e) {
+            throw logger.logExceptionAsWarning(new RuntimeException(e));
+        }
+        return Mono.empty();
+    }
+
+    @Override
+    public Mono<Void> globalSetupAsync() {
+        // Since test does warm up and test many times, we are sending many messages, so we will have them available.
+        int totalMessageMultiplier = 50;
+
+        String messageId = UUID.randomUUID().toString();
+        Message message = new Message(CONTENTS);
+        message.setMessageId(messageId);
+
+        return Mono.defer(() -> {
+            messages = IntStream.range(0, options.getMessagesToSend() * totalMessageMultiplier)
+                .mapToObj(index -> message)
+                .collect(Collectors.toList());
+
+            try {
+                sender.sendBatch(messages);
```

Comment:
Use the async option. sendBatchAsync. There is a static method that changes it to the appropriate reactor type